### ISSUE TITLE
FixedPointNumber bug

### ISSUE THREAD

**Issue by a7853z** [FederatedAI/FATE#3687]
Find a bug about FixedPointNumber by following code:
```
from federatedml.secureprotol.fixedpoint import FixedPointNumber
a = FixedPointNumber(encoding=63172158976443095, exponent=10, n=2305843009213693952)
a.decode()
Out[59]: 57454.743888632125
b = FixedPointNumber(encoding=318909341158290850, exponent=13, n=2305843009213693952)
b.decode()
Out[61]: 70.81209866439472
c=a+b
c.decode()
Out[63]: 181.55598729651896
```

The result for c.decode is obviously wrong. Another example:
```
n=2<<60
b_enc = FixedPointNumber.encode(5.4388888888888888888888, n=n, max_int= n//2)
b_enc = FixedPointNumber.encode(0.0000043888888888, n=n, max_int= n//2)
c_enc=a_enc+b_enc
c_enc.decode()
Out[56]: -0.0018893811111099956
```
Also, cann't get a correct result of fate1.7's HeteroPearson while fate1.6 can.  I was trying to locate the bug, then find this bug.

**Comment by dylan-fan** [comment]
HeteroPearson problem is fixed in 1.7.1.  Please read the readme file for 1.7.1

**Comment by a7853z** [comment]
> HeteroPearson problem is fixed in 1.7.1. Please read the readme file for 1.7.1

Sorry , but I did ran the test with 1.7.1, so please check it again and run the above code from your side and confirm this bug.

**Comment by dylan-fan** [comment]
There are two problems here.
1. The problem of HeteroPearson was triggered by a bug in spdz. The bug was fixed in 1.7.1. You can check HeteroPearson result. IF any problem , please tell us.

2.  The case of FixedPointNumber above code is because of the precision overflow

**Comment by dylan-fan** [comment]
n=2<<100
a_enc = FixedPointNumber.encode(5.4388888888888888888888, n=n, max_int= n//2)
b_enc = FixedPointNumber.encode(0.0000043888888888, n=n, max_int= n//2)
c_enc=a_enc+b_enc
c_enc.decode()

**Comment by dylan-fan** [comment]
n=2<<60
a_enc = FixedPointNumber.encode(5.43, n=n, max_int= n//2)
b_enc = FixedPointNumber.encode(0.01, n=n, max_int= n//2)
c_enc=a_enc+b_enc
c_enc.decode()

**Comment by a7853z** [comment]
> There are two problems here.
> 
> 1. The problem of HeteroPearson was triggered by a bug in spdz. The bug was fixed in 1.7.1. You can check HeteroPearson result. IF any problem , please tell us.
> 2. The case of FixedPointNumber above code is because of the precision overflow

Yes, I do have a problem with SPDZ. I ran some tests of hetero_pearson with breast_hetero_guest and breast_hetero_host.  If I calculate the cross corr matrix in plaintext, the first row of this matrix is:

```
[ 0.96953704  0.35257222  0.96947443  0.96274416  0.21311972  0.53531431
6541
  0.688235    0.83031597  0.18572735 -0.253691    0.71506376 -0.11169009
6542
  0.69719919  0.75737165 -0.23069028  0.20460675  0.18690313  0.35812593
6543
 -0.12812052 -0.03748756]
```
However, if I run hetero_pearson to calculate it. The first row of this matrix(before taking -1 or 1 as corr thresholds), I get a wrong result like this:
```
[-2.3295853722676476 33.08548396752294 0.6036733275926304
771
  -3.6122762742297967 -1796.6690452298524 55.3797565186993
772
  -0.04546776300293382 -551.7097829655339 -34.08303029413725
773
  3.204695583572869 109.69059633200901 -4.879499589378843
774
  -111.94774360902494 -47.04981841213806 -94.10996973508878
775
  24.815187583328814 -101.18459851591842 96.22118737589734
776
  -0.10476026461448759 47.67639461626979]
```
Also, the result varies every time I run hetero_pearson.

Please verify this from your side.
